cli/src/main/java/com/devonfw/tools/ide/environment/AbstractEnvironmentVariables.java

```diff
@@ -161,36 +163,64 @@ public EnvironmentVariables resolved() {
   @Override
   public String resolve(String string, Object src) {
 
-    return resolve(string, src, 0, src, string);
+    return resolve(string, src, 0, src, string, this);
   }
 
-  private String resolve(String value, Object src, int recursion, Object rootSrc, String rootValue) {
+  private String resolve(String value, Object src, int recursion, Object rootSrc, String rootValue,
+      AbstractEnvironmentVariables resolvedVars) {
 
     if (value == null) {
       return null;
     }
     if (recursion > MAX_RECURSION) {
-      throw new IllegalStateException("Reached maximum recursion resolving " + value + " for root valiable " + rootSrc
+      throw new IllegalStateException("Reached maximum recursion resolving " + value + " for root variable " + rootSrc
           + " with value '" + rootValue + "'.");
     }
     recursion++;
+
     Matcher matcher = VARIABLE_SYNTAX.matcher(value);
     if (!matcher.find()) {
       return value;
     }
     StringBuilder sb = new StringBuilder(value.length() + EXTRA_CAPACITY);
     do {
       String variableName = matcher.group(2);
-      String variableValue = getValue(variableName);
+      String variableValue = resolvedVars.getValue(variableName);
       if (variableValue == null) {
         this.context.warning("Undefined variable {} in '{}={}' for root '{}={}'", variableName, src, value, rootSrc,
             rootValue);
-      } else {
-        String replacement = resolve(variableValue, variableName, recursion, rootSrc, rootValue);
+        continue;
+      }
+      EnvironmentVariables lowestFound = findVariable(variableName); // for DirectoryMergerTest this returned false when
+                                                                     // resolving IDE_HOME
```

Comment:
```suggestion
      EnvironmentVariables lowestFound = findVariable(variableName);
```

Looks like a personal note. JUnit-test behavior should be documented in JUnit and not in productive code. At this point for other developers this is more confusing than helpful. Besides "false" does not make sense to me in this context at all, so I assume it is an outdated leftover.